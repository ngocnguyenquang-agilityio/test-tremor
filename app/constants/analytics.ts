/* Constants for Analytics Page */
export const ANALYTICS_STATISTICAL_TYPE = {
  BOOKINGS: "Bookings",
  REVENUE: "Revenue",
  TODAY_USER: "Today's Users",
  FOLLOWERS: "Followers",
};

export const enum CHART_TYPE {
  SALE = "sale",
  PERFORMANCE = "performance",
}
