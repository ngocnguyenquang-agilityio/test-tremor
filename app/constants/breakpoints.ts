export const BREAKPOINTS = {
  MD: 768,
  LG: 1024,
};
