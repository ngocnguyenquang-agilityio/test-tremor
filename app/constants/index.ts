export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./sections";
export * from "./images";
