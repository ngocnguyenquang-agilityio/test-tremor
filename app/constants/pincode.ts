const DEFAULT_PIN_CODE = "*";
const PIN_CODE_LENGTH = 6;

export { DEFAULT_PIN_CODE, PIN_CODE_LENGTH };
