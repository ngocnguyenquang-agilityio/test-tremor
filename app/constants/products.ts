// Types
import { SelectOptionData } from "@/types";

type TopSellingProductsType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
  5: string;
};

export const TOP_SELLING_PRODUCTS_SRC: TopSellingProductsType = {
  1: "/assets/images/products/blue-shoe.webp",
  2: "/assets/images/products/black-mug.webp",
  3: "/assets/images/products/black-chair.webp",
  4: "/assets/images/products/bang-sound.webp",
  5: "/assets/images/products/photo-tools.webp",
};

export const TYPE_PRICE: SelectOptionData[] = [
  {
    option: "BTC",
    value: "1",
  },
  {
    option: "CNY",
    value: "2",
  },
  {
    option: "EUR",
    value: "3",
  },
  {
    option: "GBP",
    value: "4",
  },
  {
    option: "INR",
    value: "5",
  },
  {
    option: "USD",
    value: "6",
  },
];

export const TAGS_PRICE: SelectOptionData[] = [
  {
    option: "Black Friday",
    value: "1",
  },
  {
    option: "Expired",
    value: "2",
  },
  {
    option: "Out of Stock",
    value: "3",
  },
  {
    option: "In Stock",
    value: "4",
  },
  {
    option: "Sale",
    value: "5",
  },
];

export const CATEGORY_PRODUCT: SelectOptionData[] = [
  {
    option: "Clothing",
    value: "1",
  },
  {
    option: "Electronics",
    value: "2",
  },
  {
    option: "Furniture",
    value: "3",
  },
  {
    option: "Others",
    value: "4",
  },
  {
    option: "Real Estate",
    value: "5",
  },
];

export const SIZE_PRODUCT: SelectOptionData[] = [
  {
    option: "Extra Large",
    value: "1",
  },
  {
    option: "Extra Small",
    value: "2",
  },
  {
    option: "Large",
    value: "3",
  },
  {
    option: "Medium",
    value: "4",
  },
  {
    option: "Small",
    value: "5",
  },
];
