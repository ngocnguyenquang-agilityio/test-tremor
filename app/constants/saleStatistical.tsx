import { ReactNode } from "react";
import { IoMdPersonAdd } from "react-icons/io";
import { MdLeaderboard, MdStore, MdWeekend } from "react-icons/md";

interface Icon {
  bgIcon: string;
  icon: ReactNode;
}

interface Sale {
  [key: string]: Icon;
}

export const SALE_STATISTICAL: Sale = {
  bookings: {
    bgIcon: "bg-gradient-arsenic",
    icon: <MdWeekend color="white" size="24px" />,
  },
  todayUser: {
    bgIcon: "bg-[linear-gradient(195deg,#49a3f1,#1A73E8)]",
    icon: <MdLeaderboard color="white" size="24px" />,
  },
  revenue: {
    bgIcon: "bg-[linear-gradient(195deg,#66BB6A,#43A047)]",
    icon: <MdStore color="white" size="24px" />,
  },
  followers: {
    bgIcon: "bg-[linear-gradient(195deg,#EC407A,#D81B60)]",
    icon: <IoMdPersonAdd color="white" size="24px" />,
  },
};
