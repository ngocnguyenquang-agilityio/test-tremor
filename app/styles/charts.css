.recharts-dot.recharts-line-dot {
  @apply fill-[azure];
  stroke-opacity: inherit;
}

.recharts-text.recharts-cartesian-axis-tick-value {
  @apply stroke-none fill-[white] font-thin text-sm;
}

.analytics-line-chart .recharts-curve.recharts-line-curve,
.recharts-dot.recharts-line-dot {
  @apply stroke-white font-thin;
}

.recharts-curve.recharts-line-curve,
.recharts-dot.recharts-line-dot {
  @apply stroke-[3];
}

.recharts-pie {
  @apply stroke-2;
}
