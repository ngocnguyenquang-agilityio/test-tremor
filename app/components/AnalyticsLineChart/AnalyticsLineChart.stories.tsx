import type { Meta, StoryObj } from "@storybook/react";

// Components
import AnalyticsLineChart from "./AnalyticsLineChart";

// Mocks
import { LINE_CHART_DATA } from "@/mocks/charts";

//Types
import { CHART_TYPE } from "@/constants";

const meta = {
  title: "Components/AnalyticsLineChart",
  component: AnalyticsLineChart,
  tags: ["autodocs"],
} as Meta<typeof AnalyticsLineChart>;

export default meta;

type Story = StoryObj<typeof meta>;

export const AnalyticsPerformanceLineChart: Story = {
  render: () => (
    <div className="mt-10">
      <AnalyticsLineChart
        dataChart={LINE_CHART_DATA[0].data}
        type={CHART_TYPE.PERFORMANCE}
        title="Completed Tasks"
        subTitle={"Last Campaign Performance"}
        scheduleText={"just updated"}
        isDailyChart={false}
      />
    </div>
  ),
};

export const AnalyticsSaleLineChart: Story = {
  render: () => (
    <div className="mt-10">
      <AnalyticsLineChart
        dataChart={LINE_CHART_DATA[1].data}
        type={CHART_TYPE.SALE}
        title={"Daily Sales"}
        descValue="(+15%)"
        subTitle={"increase in today sales."}
        scheduleText={"updated 4 mins ago"}
        isDailyChart
      />
    </div>
  ),
};
