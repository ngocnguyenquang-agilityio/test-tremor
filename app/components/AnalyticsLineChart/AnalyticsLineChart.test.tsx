import { render } from "@testing-library/react";

// Components
import AnalyticsLineChart from "./AnalyticsLineChart";

// Types
import { CHART_TYPE } from "@/constants";

// Mocks
import { LINE_CHART_DATA } from "@/mocks/charts";

describe("Testing AnalyticsLineChart component", () => {
  window.ResizeObserver =
    window.ResizeObserver ||
    jest.fn().mockImplementation(() => ({
      disconnect: jest.fn(),
      observe: jest.fn(),
      unobserve: jest.fn(),
    }));

  const propsDefault = {
    dataChart: LINE_CHART_DATA[1].data,
    type: CHART_TYPE.SALE,
    title: "Daily Sales",
    descValue: "(+15%)",
    subTitle: "increase in today sales.",
    scheduleText: "updated 4 mins ago",
  };

  it("Should match snapshot", () => {
    const { container } = render(
      <AnalyticsLineChart isDailyChart={false} {...propsDefault} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render correctly title with display AnalyticsLineChart", () => {
    const { getByText } = render(
      <AnalyticsLineChart
        dataChart={LINE_CHART_DATA[0].data}
        type={CHART_TYPE.PERFORMANCE}
        title="Completed Tasks"
        subTitle={"Last Campaign Performance"}
        scheduleText={"just updated"}
        isDailyChart={false}
      />,
    );

    const title = getByText("Completed Tasks");

    expect(title).toBeTruthy;
  });
});
