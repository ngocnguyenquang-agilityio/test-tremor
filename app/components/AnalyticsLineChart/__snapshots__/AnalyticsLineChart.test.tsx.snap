// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AnalyticsLineChart component Should match snapshot 1`] = `
<div>
  <div
    class="w-full bg-white bg-clip-border shadow-line-chart"
  >
    <div
      class="p-4"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <main
          class="w-full mx-auto"
        >
          <div
            class="-mt-10"
          >
            <div
              class="tremor-Card-root relative w-full text-left ring-1 rounded-tremor-default ring-tremor-ring shadow-tremor-card dark:bg-dark-tremor-background dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand bg-[linear-gradient(195deg,rgb(102,187,106),rgb(67,160,71))] p-2 text-[rgb(52,71,103)] rounded-lg shadow-[rgba(0,0,0,0.14)_0rem_0.25rem_1.25rem_0rem,rgba(76,175,79,0.4)_0rem_0.4375rem_0.625rem_-0.3125rem]"
            >
              <div
                class="w-full h-[168px] mt-4"
              >
                <div>
                  <div
                    class="recharts-wrapper"
                    role="region"
                    style="position: relative; cursor: default; width: 100px; height: 100px;"
                  >
                    <svg
                      class="recharts-surface"
                      height="100"
                      style="width: 100%; height: 100%;"
                      viewBox="0 0 100 100"
                      width="100"
                    >
                      <title />
                      <desc />
                      <defs>
                        <clippath
                          id="recharts1-clip"
                        >
                          <rect
                            height="60"
                            width="60"
                            x="35"
                            y="5"
                          />
                        </clippath>
                      </defs>
                      <g
                        class="recharts-cartesian-grid"
                      >
                        <g
                          class="recharts-cartesian-grid-horizontal"
                        >
                          <line
                            class="stroke-1 stroke-tremor-border dark:stroke-dark-tremor-border"
                            fill="none"
                            height="60"
                            stroke="#ccc"
                            width="60"
                            x="35"
                            x1="35"
                            x2="95"
                            y="5"
                            y1="65"
                            y2="65"
                          />
                          <line
                            class="stroke-1 stroke-tremor-border dark:stroke-dark-tremor-border"
                            fill="none"
                            height="60"
                            stroke="#ccc"
                            width="60"
                            x="35"
                            x1="35"
                            x2="95"
                            y="5"
                            y1="50"
                            y2="50"
                          />
                          <line
                            class="stroke-1 stroke-tremor-border dark:stroke-dark-tremor-border"
                            fill="none"
                            height="60"
                            stroke="#ccc"
                            width="60"
                            x="35"
                            x1="35"
                            x2="95"
                            y="5"
                            y1="35"
                            y2="35"
                          />
                          <line
                            class="stroke-1 stroke-tremor-border dark:stroke-dark-tremor-border"
                            fill="none"
                            height="60"
                            stroke="#ccc"
                            width="60"
                            x="35"
                            x1="35"
                            x2="95"
                            y="5"
                            y1="20"
                            y2="20"
                          />
                          <line
                            class="stroke-1 stroke-tremor-border dark:stroke-dark-tremor-border"
                            fill="none"
                            height="60"
                            stroke="#ccc"
                            width="60"
                            x="35"
                            x1="35"
                            x2="95"
                            y="5"
                            y1="5"
                            y2="5"
                          />
                        </g>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis text-tremor-label fill-tremor-content dark:fill-dark-tremor-content"
                      >
                        <g
                          class="recharts-cartesian-axis-ticks"
                        >
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="30"
                              orientation="bottom"
                              stroke="none"
                              text-anchor="middle"
                              transform="translate(0, 6)"
                              width="60"
                              x="55"
                              y="73"
                            >
                              <tspan
                                dy="0.71em"
                                x="55"
                              >
                                Apr
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="30"
                              orientation="bottom"
                              stroke="none"
                              text-anchor="middle"
                              transform="translate(0, 6)"
                              width="60"
                              x="60"
                              y="73"
                            >
                              <tspan
                                dy="0.71em"
                                x="60"
                              >
                                Jun
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="30"
                              orientation="bottom"
                              stroke="none"
                              text-anchor="middle"
                              transform="translate(0, 6)"
                              width="60"
                              x="65"
                              y="73"
                            >
                              <tspan
                                dy="0.71em"
                                x="65"
                              >
                                Aug
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="30"
                              orientation="bottom"
                              stroke="none"
                              text-anchor="middle"
                              transform="translate(0, 6)"
                              width="60"
                              x="70"
                              y="73"
                            >
                              <tspan
                                dy="0.71em"
                                x="70"
                              >
                                Oct
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="30"
                              orientation="bottom"
                              stroke="none"
                              text-anchor="middle"
                              transform="translate(0, 6)"
                              width="60"
                              x="75"
                              y="73"
                            >
                              <tspan
                                dy="0.71em"
                                x="75"
                              >
                                Dec
                              </tspan>
                            </text>
                          </g>
                        </g>
                      </g>
                      <g
                        class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis text-tremor-label fill-tremor-content dark:fill-dark-tremor-content"
                      >
                        <g
                          class="recharts-cartesian-axis-ticks"
                        >
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="60"
                              orientation="left"
                              stroke="none"
                              text-anchor="end"
                              transform="translate(-3, 0)"
                              width="30"
                              x="27"
                              y="65"
                            >
                              <tspan
                                dy="0.355em"
                                x="27"
                              >
                                0
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="60"
                              orientation="left"
                              stroke="none"
                              text-anchor="end"
                              transform="translate(-3, 0)"
                              width="30"
                              x="27"
                              y="50"
                            >
                              <tspan
                                dy="0.355em"
                                x="27"
                              >
                                150
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="60"
                              orientation="left"
                              stroke="none"
                              text-anchor="end"
                              transform="translate(-3, 0)"
                              width="30"
                              x="27"
                              y="35"
                            >
                              <tspan
                                dy="0.355em"
                                x="27"
                              >
                                300
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="60"
                              orientation="left"
                              stroke="none"
                              text-anchor="end"
                              transform="translate(-3, 0)"
                              width="30"
                              x="27"
                              y="20"
                            >
                              <tspan
                                dy="0.355em"
                                x="27"
                              >
                                450
                              </tspan>
                            </text>
                          </g>
                          <g
                            class="recharts-layer recharts-cartesian-axis-tick"
                          >
                            <text
                              class="recharts-text recharts-cartesian-axis-tick-value"
                              fill=""
                              height="60"
                              orientation="left"
                              stroke="none"
                              text-anchor="end"
                              transform="translate(-3, 0)"
                              width="30"
                              x="27"
                              y="5"
                            >
                              <tspan
                                dy="0.355em"
                                x="27"
                              >
                                600
                              </tspan>
                            </text>
                          </g>
                        </g>
                      </g>
                      <g
                        class="recharts-layer recharts-line stroke-blue-500"
                      >
                        <path
                          class="recharts-curve recharts-line-curve"
                          d="M55,59.99999999999999L57.5,61L60,35L62.5,33L65,14.999999999999996L67.5,30L70,45L72.5,42L75,14.999999999999996"
                          fill="none"
                          height="60"
                          name="Mobile apps"
                          stroke=""
                          stroke-dasharray="0px 0px"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                          stroke-opacity="1"
                          stroke-width="2"
                          width="60"
                        />
                      </g>
                      <g
                        class="recharts-layer recharts-line cursor-pointer"
                      >
                        <path
                          class="recharts-curve recharts-line-curve"
                          d="M55,59.99999999999999L57.5,61L60,35L62.5,33L65,14.999999999999996L67.5,30L70,45L72.5,42L75,14.999999999999996"
                          fill="none"
                          height="60"
                          name="Mobile apps"
                          stroke="transparent"
                          stroke-dasharray="0px 0px"
                          stroke-opacity="0"
                          stroke-width="12"
                          width="60"
                        />
                      </g>
                    </svg>
                    <div
                      class="recharts-legend-wrapper"
                      style="position: absolute; width: 90px; height: 60px; left: 5px; top: 5px;"
                    >
                      <div
                        class="flex items-center justify-end"
                      >
                        <ol
                          class="tremor-Legend-root relative overflow-hidden"
                        >
                          <div
                            class="h-full flex flex-wrap"
                            tabindex="0"
                          >
                            <li
                              class="tremor-Legend-legendItem group inline-flex items-center px-2 py-0.5 rounded-tremor-small transition whitespace-nowrap cursor-pointer text-tremor-content hover:bg-tremor-background-subtle dark:text-dark-tremor-content dark:hover:bg-dark-tremor-background-subtle"
                            >
                              <svg
                                class="flex-none text-blue-500 h-2 w-2 mr-1.5 opacity-100"
                                fill="currentColor"
                                viewBox="0 0 8 8"
                              >
                                <circle
                                  cx="4"
                                  cy="4"
                                  r="4"
                                />
                              </svg>
                              <p
                                class="whitespace-nowrap truncate text-tremor-default text-tremor-content group-hover:text-tremor-content-emphasis dark:text-dark-tremor-content opacity-100 dark:group-hover:text-dark-tremor-content-emphasis"
                              >
                                Mobile apps
                              </p>
                            </li>
                          </div>
                        </ol>
                      </div>
                    </div>
                    <div
                      class="recharts-tooltip-wrapper"
                      role="dialog"
                      style="visibility: hidden; pointer-events: none; position: absolute; top: 0px; left: 0px; outline: none;"
                      tabindex="-1"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
      <div
        class="flex-col items-start pt-6 pb-2 px-2"
      >
        <div>
          <p
            class="text-tremor-title dark:text-dark-tremor-content-emphasis text-base font-bold opacity-100 capitalize text-[#344767]"
          >
            Daily Sales
          </p>
          <div
            class="w-fit h-fit relative flex justify-center"
          >
            <p
              class="dark:text-dark-tremor-content-subtle text-sm font-light opacity-100 text-[#7b809a]"
            >
              <span
                class="font-bold"
              >
                (+15%)
              </span>
               
              increase in today sales.
            </p>
            <div
              class="min-w-fit w-[200px] h-fit absolute bottom-[100%] z-50 transition-all"
              hidden=""
            >
              <div
                class="text-sm font-light text-secondary rounded bg-white p-3 shadow-md mb-[10px]"
              >
                increase in today sales.
              </div>
            </div>
          </div>
        </div>
        <hr
          class="bg-gradient-line h-px opacity-25 mx-0 my-4 border-b-[none] border-solid"
        />
        <div
          class="tremor-Flex-root flex w-full flex-row items-center justify-start"
        >
          <svg
            color="#7b809a"
            fill="currentColor"
            height="12"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(123, 128, 154);"
            viewBox="0 0 512 512"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm61.8-104.4l-84.9-61.7c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v141.7l66.8 48.6c5.4 3.9 6.5 11.4 2.6 16.8L334.6 349c-3.9 5.3-11.4 6.5-16.8 2.6z"
            />
          </svg>
          <p
            class="text-tremor-default dark:text-dark-tremor-content text-sm font-light opacity-100 text-[#7b809a] ml-1"
          >
            updated 4 mins ago
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
